docs/canonical_testing.md

```diff
@@ -0,0 +1,186 @@
+# Canonical testing overview
+
+We want to make it trivial for everyone to discover how to run tests, and to run
+tests the same way.
+
+In order to do this we will provide:
+
+* A new standardized command to invoke any test job, which calls
+  - `k8stest`
+  - This knows how to map jobs into a job class.
```

Comment:
Similar actions -- build, verify, unit, integration, e2e, node e2e, kubelet, etc

Basically if the same test is run multiple times with different args/variables it is considered a class.

So `kubernetes-test-go` and `kubernetes-e2e-gce` are different classes (unit and e2e). `kubernetes-e2e-gce` and `kubernetes-e2e-gke-serial` are the same class (both e2e).